examples/tv-casting-app/darwin/MatterTvCastingBridge/MatterTvCastingBridge/CastingServerBridge.mm

```diff
@@ -118,10 +109,157 @@ - (instancetype)init
     return self;
 }
 
+/**
+ Dispatches the block on the Matter SDK dispatch queue.
+
+ @param description Optional. A description of the callback to be dispatched for diagnostic purposes.
+ @param block The Block to be dispatched.
+ */
+- (void) dispatchOnMatterSDKQueue:(const NSString * _Nullable)description
+                            block:(dispatch_block_t)block
+{
+    if (nil != description) {
+        ChipLogProgress(AppServer, "[SYNC] CastingServerBridge %s", [description UTF8String]);
+    }
+    dispatch_async(_chipWorkQueue, ^{
+        // The Matter SDK is not, generally, internally thread-safe. It relies upon the client to lock
+        // the SDK prior to any interactions after chip::DeviceLayer::PlatformMgr().InitChipStack()
+        //
+        // Note that it is presently safe to do at this point because InitChipStack is called in the
+        // constructor for the CastingServerBridge.
+        chip::DeviceLayer::StackLock lock;
```

Comment:
The client callbacks are dispatched asynchronously on a separate dispatch queue, so if the client callback invokes API 2 in response to a callback from API 1, API 1's callback will already, synchronously, have finished its block and released the stack lock. This is the pattern that we'll want to eventually bring over to the Android side as well because it saves the `chip::DeviceLayer::StackUnlock`s.

I can add an example usage to the sample app.